clients/pkg/promtail/promtail.go

```diff
@@ -139,10 +140,16 @@ func (p *Promtail) reloadConfig(cfg *config.Config) error {
 	}
 
 	cfg.Setup(p.logger)
+	var err error
+	err = scrapeconfig.ValidateJobName(cfg.ScrapeConfig)
```

Comment:
Do we only need the check during a config reload? Is it not also necessary when Promtail starts?